qt/DetailsDialog.cc

```diff
@@ -1289,38 +1290,64 @@ void DetailsDialog::onTrackerSelectionChanged()
 void DetailsDialog::onAddTrackerClicked()
 {
     bool ok = false;
-    QString const url = QInputDialog::getText(this, tr("Add URL "), tr("Add tracker announce URL:"), QLineEdit::Normal,
-        QString(), &ok);
+    QString const urls = QInputDialog::getMultiLineText(this, tr("Add URL(s) "),
+        tr("Add tracker announce URL(s) (one per line):"), {}, &ok);
+    QStringList urlList = urls.split(QRegExp(QString::fromStdString("\n|\r\n|\r")), QString::SkipEmptyParts);
+
+    // Filter out invalid urls
+    urlList.erase(std::remove_if(urlList.begin(),
+        urlList.end(),
+        [](auto const& url) { return !QUrl(url).isValid(); }),
+        urlList.end());
+
+    // Remove duplicate urls
+    urlList.sort();
+    urlList.erase(std::unique(urlList.begin(), urlList.end()),
+        urlList.end());
```

Comment:
* The regex can be simplified to `[\r\n]+`
 * Slightly more efficient to use `QStringLiteral` instead of `QString::fromStdString()`
 * If we keep the temporary QUrl objects, the uniqueness check will be more robust
 * If we use `QSet<QUrl>` the code will be simpler because uniqueness will be enforced automatically

```suggestion
    QSet<QUrl> urls;
    for (auto const& line : text.split(QRegExp(QStringLiteral("[\r\n]+"))))
    {
        QUrl const url(line.trimmed());
        if (url.isValid())
        {
            urls.insert(url);
        }
    }
```